lib/algos/list/delimiterMaster.js

```diff
@@ -120,7 +135,24 @@ class DelimiterMaster extends Delimiter {
         return this.addContents(key, value);
     }
 
+    filterV1(obj) {
+        // Filtering master keys in v1 is simply listing the master
+        // keys, as the state of version keys do not change the
+        // result, so we can use Delimiter method directly.
+        return super.filter(obj);
+    }
+
     skipping() {
+        if (this.vFormat === BucketVersioningKeyFormat.v0) {
```

Comment:
Just want to mention the idea of having each version be either a class or an object instead of putting a bunch of these if's around. Could have just one if in a factory method somewhere then everything else is standard.
Mabye a base Bucket Version class if there is a lot of similarities.
Alternatively, an object that maps the v0 and v1 methods around in one place.
Something along the lines of: ```
const versionRouter = {
  [BucketVersioningKeyFormat.v0]: {
    skipping: this.skippingV0,
    other: this.otherv0,
  },
  [BucketVersioningKeyFormat.v1]: { etc }
}

// then map through this:
versionRouter[this.vFormat].skipping()
// make wrap it with a check the version exists if required
```

Anyway, just ideas.

Reply:
It sounds better indeed since there is a common pattern, especially after the migration states are introduced (v0 to v1)

Reply:
updated